spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/support/GcpHeaders.java

```diff
@@ -0,0 +1,30 @@
+/*
+ *  Copyright 2017 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *  
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *  
+ */
+
+package org.springframework.cloud.gcp.pubsub.support;
+
+/**
+ * Google Cloud Platform internal headers for Spring Messaging messages.
+ *
+ * @author João André Martins
+ */
+public abstract class GcpHeaders {
+
+	private static final String PREFIX = "gcp_";
+
+	public static final String CONSUMER = PREFIX + "pub_sub_consumer";
```

Comment:
I told already: `ACKNOWLEDGMENT` and therefore constant value `PREFIX + "pub_sub_acknowledgment"`. That's fully reflect what we have in Kafka: https://github.com/spring-projects/spring-kafka/blob/master/spring-kafka/src/main/java/org/springframework/kafka/support/converter/MessagingMessageConverter.java#L83

Plus we should consider an `ackMode` - `auto/manual` and don't add this header in case of `auto`. 
Different story from this PR, but should be on radar as an improvement.